packages/dendron-cli/src/commands/notes.ts

```diff
@@ -205,31 +206,73 @@ export class NoteCLICommand extends CLICommand<CommandOpts, CommandOutput> {
         }
         case NoteCommands.LOOKUP_LEGACY: {
           const { query, vault } = checkQueryAndVault(opts);
-          const { data } = await engine.getNoteByPath({
-            npath: query,
-            createIfNew: true,
-            vault,
-          });
-          let stringOutput: string = "";
-          if (data?.note) {
-            stringOutput = await formatNotes({
+          const notes = await engine.findNotes({ fname: query, vault });
+          let note: NoteProps;
+
+          // If note doesn't exist, create note with schema
+          if (notes.length === 0) {
+            note = NoteUtils.createWithSchema({
+              noteOpts: {
+                fname: query,
+                vault,
+              },
               engine,
-              notes: [data.note],
-              output,
             });
-            this.print(stringOutput);
-            return {
-              data: {
-                notesOutput: [data.note],
-                stringOutput,
-              } as NoteCommandData,
-            };
+            // Until we support user prompt, pick template note for them if there are multiple matches in order of
+            // 1. Template note that lies in same vault as note to lookup
+            // 2. First note in list
+            await TemplateUtils.findAndApplyTemplate({
+              note,
+              engine,
+              pickNote: async (choices: NoteProps[]) => {
+                const sameVaultNote = choices.filter((ent) => {
```

Comment:
nitpick: Could use `choices.find` since we're trying to find only one matching

Reply:
good point, fixed